src/app/config/newTeam/newTeamReducer.js

```diff
@@ -0,0 +1,51 @@
+import { GET_USERS_REQUEST_SUCCESS, NEW_TEAM_ADD_USER, NEW_TEAM_REMOVE_USER_FROM, NEW_TEAM_RESET, NEW_TEAM_UNSAVED_CHANGES } from "../constants";
+
+export const initialState = {
+    usersInTeam: [],
+    usersNotInTeam: [],
+    allUsers: [],
+    unsavedChanges: false,
+};
+
+const newTeamReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case GET_USERS_REQUEST_SUCCESS: {
+            return {
+                ...state,
+                allUsers: [...action.users],
+                usersNotInTeam: [...action.users],
+            };
+        }
+        case NEW_TEAM_UNSAVED_CHANGES: {
+            return {
+                ...state,
+                unsavedChanges: action.unsavedChanges,
+            };
+        }
+        case NEW_TEAM_REMOVE_USER_FROM: {
+            return {
+                ...state,
+                usersInTeam: state.usersInTeam.filter(filteredUser => filteredUser.email !== action.user.email),
+                usersNotInTeam: [...state.usersNotInTeam, action.user],
+            };
+        }
+        case NEW_TEAM_ADD_USER: {
+            return {
+                ...state,
+                usersNotInTeam: state.usersNotInTeam.filter(filteredUser => filteredUser.email !== action.user.email),
```

Comment:
those values should be done in selectors not in reducer. I don't think other places in app uses this value
usersNotInTeam and usersInTeam should be derived state

Reply:
I don't think those objects are being used outside this functionality and any other component is subscribed to those changes, where you would need to use props drilling.
in situations like this you just need to calculate two separate lists based on one state in store `allUsers` , as at this is the only source of true about who belongs where.